tests/integration/test_db_router.py

```diff
@@ -181,7 +181,9 @@ async def test_keystone_bundle_db_router(ops_test: OpsTest) -> None:
     # Reduce the update_status frequency for the duration of the test
     async with ops_test.fast_forward():
         # Wait until the mysql charm is successfully deployed
-        await ops_test.model.block_until(lambda: len(ops_test.model.applications[APP_NAME].units) == 3)
+        await ops_test.model.block_until(
+            lambda: len(ops_test.model.applications[APP_NAME].units) == 3
```

Comment:
You can check the exact units during your `wait_for_idle` call with the `wait_for_exact_units` parameter